File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/CalcHurtValueMessage.ts
import { IMessage } from "../../../../Core/ObserverMessageKit";
import DamageInfo from "../DataDefine/DamageInfo";
import AbstractCreature from "../Unit/AbstractCreature";

//计算伤害
export class CalcHurtValueMessage implements  IMessage{
    public static readonly Id = "CalcDamageValueMessage";
    public constructor(public info:DamageInfo,public source:AbstractCreature,public target: AbstractCreature) {
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/CalcHurtValueMessage.cs
using ModTheSpire.DataDefine;
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///计算受到伤害
    public class CalcHurtValueMessage: IEventHandle
    {
        public DamageInfo DamageInfo;
        public AbstractCreature Source;
        public AbstractCreature Target;

        public CalcHurtValueMessage(DamageInfo damageInfo, AbstractCreature source, AbstractCreature target)
        {
            DamageInfo = damageInfo;
            Source = source;
            Target = target;
        }
    }
}